content/2017.04.getting-started-with-elm.rst

```diff
@@ -0,0 +1,487 @@
+
+
+Getting started
+===============
+
+Installing Elm
+--------------
+
+First of all install ``elm`` and ``elm-format`` using npm the node
+package manager:
+
+.. code-block:: bash
+
+    npm install -g elm elm-format
+
+What is npm? How to install it?
+-------------------------------
+
+npm is the node package manager. To install it, you might want to use
+`nvm <https://github.com/creationix/nvm#install-script>`_.
+
+Once nvm is installed, you can use::
+
+    nvm install node
+
+And then you wil be able to run the previous ``npm`` command.
+
+Looking for an IDE?
+-------------------
+
+Have a look at `Atom <https://atom.io/>`. By default it will run
+``elm-format`` each time you save your file which is really handy.
+
+Take a few moments to make sure your text editor is configured to work
+friendly with Elm files.
+
+
+Starting your first project
+===========================
+
+No need for a boilerplate here, you can just start by running
+``elm-package install`` and approve the plan:
+
+.. code-block:: bash
+
+    ~/tutorial$ elm-package install
+
+    Some new packages are needed. Here is the upgrade plan.
+
+      Install:
+        elm-lang/core 5.1.1
+        elm-lang/html 2.0.0
+        elm-lang/virtual-dom 2.0.4
+
+    Do you approve of this plan? [Y/n] Y
+    Starting downloads...
+
+      ● elm-lang/html 2.0.0
+      ● elm-lang/virtual-dom 2.0.4
+      ● elm-lang/core 5.1.1
+
+    Packages configured successfully!
+
+You now have a ``elm-package.json`` in your project as well as a
+``elm-stuff/`` directory that contains libraries ``elm-package``
+installed.
+
+.. code-block:: bash
+
+    ~/tutorial$ tree -L2
+	.
+    ├── elm-package.json
+    └── elm-stuff
+        ├── exact-dependencies.json
+        └── packages
+
+    2 directories, 2 files
+
+
+Creating your first file
+========================
+
+No need for a boilerplate here, to get started you can simply create a
+new file named ``Main.elm`` with the following:
+
+.. code-block:: elm
+
+    import Html
+    main = Html.text "Hello world"
+
+Playing with elm-format
+-----------------------
+
+If your editor is well configured with ``elm-format``, you should see
+this as soon as you save it:
+
+.. code-block:: elm
+
+    module Main exposing (..)
+
+    import Html
+
+
+    main =
+        Html.text "Hello world!"
+
+If not, you can run ``elm-format`` manually on your file:
+
+.. code-block:: bash
+
+    ~/tutorial$ elm-format Main.elm
+    This will overwrite the following files to use Elm's preferred style:
+
+        Main.elm
+
+    This cannot be undone! Make sure to back up these files before proceeding.
+
+    Are you sure you want to overwrite these files with formatted versions? (y/n)
+
+    y
+    Processing file Main.elm
+
+To avoid having to answer yes each time you can use the ``--yes`` parameter::
+
+    elm-format --yes Main.elm
+
+
+Opening your app in the browser
+-------------------------------
+
+One way to run your app is to use ``elm-reactor``.
```

Comment:
It's strange we never introduced elm-reactor while we did so with elm-live, readers may genuinely wonder why we didn't use elm-live directly.

Maybe we should just introduce elm-live later in the post, in some *Niceties* section?